drugs.com/drug-search.py

```diff
@@ -0,0 +1,216 @@
+	for item in drug.select('.nav-item'):
+		if ('side effects' == item.string.lower()):
+			sfx_link = 'https://www.drugs.com' + item.get('href')
+			break;
+	sfxsoup = make_soup(sfx_link)
+	firsth3 = sfxsoup.find('h3')
+	uls = []
+	for item in firsth3.find_next_siblings():
+		if item.name == 'h3' or item.name == 'div':
+			break
+		if item.name == 'ul':
+			uls.append(item)
+	lis = []
+	for ul in uls:
+		for li in ul.find_all('li'):
+			if li.find('ul'):
+				break
+			lis.append(li.text)
+	return lis
+
+
+def createBlobInfo(genericName, brandName, dose, counter,
+                   interactions, descrip, seList, seParag,
+                   usage, indgredientList, warnings, src): 
+    d = {
+        "GenericName" : genericName,
+        "BrandName" : brandName,
+        "RecommendedDosage" : dose,
+        "OverTheCounter" : counter,
+        "Interactions" : interactions,
+        "GeneralDescriptions" : descrip,
+        "SideEffectsList" : seList,
+        "SideEffectsParag" : seParag,
+        "Usage" : usage,
+        "ActiveIngredients" : indgredientList,
+        "Warnings" : warnings,
+        "Source" : src,
+        }
+    
+    
+    return d
+
+def get_drug_page_info(drug_link):
+	if drug_link == -1:
+		return -1
+
+	page = make_soup(drug_link)
+	
+	#Getting generic name
+	names = (page.find("p", {"class" : "drug-subtitle"})).text
+	gen_index_one = names.find("Generic") + 14 
+	gen_index_two = names.find("(") - 1
+	generic_name = names[gen_index_one: gen_index_two]
+
+	#Getting brand name
+	brand_index = names.find("Brand") + 13
+
+	#Getting description paragraph
+	descrip_meta = page.findAll(attrs={"name":"description"}) 
+	descrip = (descrip_meta[0]['content'].encode('utf-8'))
+	descrip_last_index = descrip.find("Includes")
+
+	#Getting Availability
+	avail_div = page.find("div", {"class" : "drugInfoSnip drugInfoRx1"})
+
+	if avail_div != None:
+		availability = "Prescription only"
+		otc = False
+
+	elif page.find("div", {"class" : "drugInfoSnip drugInfoRx2"}) != None:
+		availability = "Over the counter"
+		otc = True
+
+	else:
+		availability = "Prescription or over the counter"
+		otc = True
+
+	#Getting warnings
+	warning_lst= page.findAll("h2")
+	warning_parag = ""
+
+	for i in warning_lst:
+		if ('Before taking' in i.text):
+			warning_before = i
+			break
+
+	for z in warning_before.find_next_siblings():
+	 	if z.name == 'h2':
+	 		break
+
+	 	elif z.name == 'p':
+	 		warning_parag += z.text
+
+	 	elif z.name == 'ul':
+	 		for li in z.find_all('li'):
+	 			warning_parag += li.text
+
+	#final vals
+	gen_name = names[gen_index_one:gen_index_two]
+	brand_name = names[brand_index:]
+	descrip = descrip[:descrip_last_index]
+
+	#warning_parag = final warning paragraph in str format
+	#otc = over the counter boolean
+	#availability = otc description (from site)
+
+	blob = createBlobInfo(gen_name, brand_name, "", otc, get_interactions(drug_link), descrip, get_sfx(drug_link), "", "", "", warning_parag, "http://www.drugs.com")
```

Comment:
A little over-engineering here, I don't think `createBlobInfo` is needed. You can just assign the dictionary to `blob` directly.